### PR TITLE
Remove required return type check for .NET Lambda handlers

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Runtime construction is valid for .NET Lambda handlers, so object is a valid return type. Since that is the case, we should not filter potential Lambda handlers based on return type.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Related Issue(s)

#1346

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-rider/ReSharper.AWS/src/AWS.Psi/Lambda/LambdaFinder.cs [hunk 1]
```diff
@@ -69,8 +69,7 @@ public static bool IsSuitableLambdaMethod(IMethod method)
             return method != null &&
                    method.GetAccessRights() == AccessRights.PUBLIC &&
                    IsValidInstanceOrStaticMethod(method) &&
-                   HasRequiredParameters(method) &&
-                   HasRequiredReturnType(method);
+                   HasRequiredParameters(method);
```
Comment (normj) [inline jetbrains-rider/ReSharper.AWS/src/AWS.Psi/Lambda/LambdaFinder.cs]: Noticed the HasRequiredParameters has a couple problems. First it is valid to have a method that takes in zero parameters. The ILambdaContext is optional as well. This is common when a user has existing code that they just want to run via a CloudWatch Event schedule.

The second issue it looks like the logic is requiring a serializer to be defined. That is not true for case where there is no input parameter or they are passing in a stream.

Comment (abrooksv) [reply jetbrains-rider/ReSharper.AWS/src/AWS.Psi/Lambda/LambdaFinder.cs]: Your comment doesnt seem to align with the docs, I read it as ILambdaContext is required in the first paragraphs.

https://docs.aws.amazon.com/lambda/latest/dg/dotnet-programming-model-handler-types.html

Can we have the doc writers fix this, please and clarify what you said is possible in the docs?

Comment (normj) [reply jetbrains-rider/ReSharper.AWS/src/AWS.Psi/Lambda/LambdaFinder.cs]: Seems like the line "If you want access to the Lambda context object, it is available by defining a method parameter of type ILambdaContext" make it sound optional.

Comment (abrooksv) [reply jetbrains-rider/ReSharper.AWS/src/AWS.Psi/Lambda/LambdaFinder.cs]: Would it not be better to be explicit like the Java page? https://docs.aws.amazon.com/lambda/latest/dg/java-programming-model-handler-types.html

`You can omit the Context object from the handler method signature if it isn't needed. For more information, see AWS Lambda Context Object in Java. `

Comment (normj) [reply jetbrains-rider/ReSharper.AWS/src/AWS.Psi/Lambda/LambdaFinder.cs]: I submitted a ticket to the Lambda doc team to be explicit like the Java page.

Comment (abrooksv) [reply jetbrains-rider/ReSharper.AWS/src/AWS.Psi/Lambda/LambdaFinder.cs]: Thanks!

**Pull request merged**